cms/utils/copy_plugins.py

```diff
@@ -7,20 +7,33 @@ def copy_plugins_to(old_plugins, to_placeholder,
     """
     Copies a list of plugins to a placeholder to a language.
     """
+    # import pdb; pdb.set_trace()
     # TODO: Refactor this and copy_plugins to cleanly separate plugin tree/node
     # copying and remove the need for the mutating parameter old_parent_cache.
     old_parent_cache = {}
-    # For subplugin copy, first plugin's parent must be nulled before copying.
+    # For subplugin copy, top-level plugin's parent must be nulled
+    # before copying.
     if old_plugins:
-        old_plugins[0].parent = old_plugins[0].parent_id = None
-    new_plugins = [old.copy_plugin(to_placeholder, to_language or old.language,
-                                   old_parent_cache, no_signals)
-                   for old in old_plugins]
+        old_parent = old_plugins[0].parent
+        for old_plugin in old_plugins:
+            if old_plugin.parent == old_parent:
+                old_plugin.parent = old_plugin.parent_id = None
+    new_plugins = []
+    for old in old_plugins:
+        new_plugins.append(
+            old.copy_plugin(to_placeholder, to_language or old.language,
+                            old_parent_cache, no_signals))
+
     if new_plugins and parent_plugin_id:
         from cms.models import CMSPlugin
-        new_plugins[0].parent_id = parent_plugin_id
-        new_plugins[0].save()
-        new_plugins[0] = new_plugins[0].move(CMSPlugin.objects.get(pk=parent_plugin_id), pos='last-child')
+        parent_plugin = CMSPlugin.objects.get(pk=parent_plugin_id)
+        top_plugins = [p for p in new_plugins if p.parent_id is None]
+        for idx, plugin in enumerate(new_plugins):
+            if plugin in top_plugins:
```

Comment:
Why not check ``if plugin.parent_id is None`` here? That might save some time for not top plugins by not walking through each element in top_plugins list...
Also in this case you don't need L#30 (``top_plugins = [p for p in new_plugins if p.parent_id is None]``)